File: templates/svix-lib-rust/api_extra/application_create.rs
/// Create the application with the given ID, or create a new one if it
/// doesn't exist yet.
pub async fn get_or_create(
    &self,
    application_in: ApplicationIn,
    options: Option<PostOptions>,
) -> Result<ApplicationOut> {
    let PostOptions { idempotency_key } = options.unwrap_or_default();

    crate::request::Request::new(http1::Method::POST, "/api/v1/app")
        .with_body_param(application_in)
        .with_query_param("get_if_exists", "true".to_owned())
        .with_optional_header_param("idempotency-key", idempotency_key)
        .execute(self.cfg)
        .await
}

File: templates/svix-lib-javascript/api_extra/application_create.ts
/** Get the application with the UID from `applicationIn`, or create it if it doesn't exist yet. */
public getOrCreate(
  applicationIn: ApplicationIn,
  options?: PostOptions
): Promise<ApplicationOut> {
  const request = new SvixRequest(HttpMethod.POST, "/api/v1/app");

  request.setQueryParam("get_if_exists", true);
  request.setHeaderParam("idempotency-key", options?.idempotencyKey);
  request.setBody(applicationIn, "ApplicationIn");

  return request.send(this.requestCtx, "ApplicationOut");
}
